File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        queue<TreeNode*>q;
        q.push(root);
        q.push(NULL);
        vector<pair<int,int>>ans;
        int res=0,idx=0;
        while(q.size()>1){
            TreeNode* temp = q.front();
            q.pop();
            if(temp==NULL){
                ++idx;
                ans.push_back(make_pair(res,idx));
                q.push(NULL);
                // cout<<res<<endl;
                res=0;
                continue;
            }
            res+=temp->val;
            // cout<<res<<endl;
            if(temp->left)q.push(temp->left);
            if(temp->right)q.push(temp->right);
        }
        ++idx;
        ans.push_back(make_pair(res,idx));
        // cout<<ans.size();
        sort(begin(ans),end(ans));
        idx=ans.size()-1;
        while(idx>0 and ans[idx].first==ans[idx-1].first) --idx;
        return ans[idx].second;
    }
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    from collections import deque
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        q = deque()
        q.append(root)
        q.append(None)
        ans = []
        res = 0
        idx = 0
        while(len(q)>1):
            temp = q.popleft()
            if(temp==None):
                idx+=1
                ans.append([res,idx])
                res=0
                q.append(None)
                continue
            res+=temp.val
            if(temp.left!=None):
                q.append(temp.left)
            if(temp.right!=None):
                q.append(temp.right)
        ans.append([res,idx+1])
        ans.sort()
        sol = []
        for i in range(len(ans)):
            if(ans[i][0]==ans[len(ans)-1][0]):
                sol.append(ans[i][1])
            
        sol.sort()
        return sol[0]